reframe/core/environments.py

```diff
@@ -266,133 +312,26 @@ def __init__(self,
-            compiler = self.cc
-            flags.append(cflags)
-        elif lang == 'C++':
-            compiler = self.cxx
-            flags.append(cxxflags)
-        elif lang == 'Fortran':
-            compiler = self.ftn
-            flags.append(fflags)
-        elif lang == 'CUDA':
-            compiler = 'nvcc'
-            flags.append(cxxflags)
-        else:
-            raise EnvironError('Unknown language: %s' % lang)
-
-        # Append include search path
-        flags += ['-I' + d for d in self.include_search_path]
-        cmd = ('%s %s %s -o %s %s %s' % (compiler, ' '.join(flags),
-                                         source_file, executable,
-                                         ldflags, options))
-        try:
-            return os_ext.run_command(cmd, check=True)
-        except SpawnedProcessError as e:
-            # Re-raise as compilation error
-            raise CompilationError(command=e.command,
-                                   stdout=e.stdout,
-                                   stderr=e.stderr,
-                                   exitcode=e.exitcode) from None
-
-    def _compile_dir(self, source_dir, makefile, options):
-        if makefile:
-            cmd = 'make -C %s -f %s %s ' % (source_dir, makefile, options)
-        else:
-            cmd = 'make -C %s %s ' % (source_dir, options)
-
-        # Pass a set of predefined options to the Makefile
-        if self.propagate:
-            flags = ["CC='%s'"  % self.cc,
-                     "CXX='%s'" % self.cxx,
-                     "FC='%s'"  % self.ftn]
-
-            # Explicitly check against None here; the user may explicitly want
-            # to clear the flags
-            if self.cppflags is not None:
-                flags.append("CPPFLAGS='%s'" % self.cppflags)
-
-            if self.cflags is not None:
-                flags.append("CFLAGS='%s'" % self.cflags)
-
-            if self.cxxflags is not None:
-                flags.append("CXXFLAGS='%s'" % self.cxxflags)
-
-            if self.fflags is not None:
-                flags.append("FFLAGS='%s'" % self.fflags)
-
-            if self.ldflags is not None:
-                flags.append("LDFLAGS='%s'" % self.ldflags)
-
-            cmd += ' '.join(flags)
-
-        try:
-            return os_ext.run_command(cmd, check=True)
-        except SpawnedProcessError as e:
-            # Re-raise as compilation error
-            raise CompilationError(command=e.command,
-                                   stdout=e.stdout,
-                                   stderr=e.stderr,
-                                   exitcode=e.exitcode) from None
+        self._cc  = cc
+        self._cxx = cxx
+        self._ftn = ftn
+        self._nvcc = nvcc
+        self._cppflags = cppflags
+        self._cflags   = cflags
+        self._cxxflags = cxxflags
+        self._fflags   = fflags
+        self._ldflags  = ldflags
+        self._include_search_path = []
+        self._propagate = True
```

Comment:
I am not treating this variable in the new build systems. I should fix that for compatibility reasons.